frontend/src/Context/SessionProvider/session.tsx

```diff
@@ -164,13 +148,23 @@ const SessionProvider = ({ children }: SessionProviderProps): ReactElement => {
     toggleReportIssue,
   } = useRightPanel();
 
-  const handleChatSignal = ({ data }: { data: string }) => {
+  const handleChatSignal = ({ data }: SignalEvent) => {
     if (data) {
       onChatMessage(data);
       incrementUnreadCount();
     }
   };
 
+  const handleEmoji = useCallback(
+    (emojiEvent: SignalEvent) => {
+      setSubscriberWrappers((currentSubscriberWrappers) => {
+        onEmoji(emojiEvent, currentSubscriberWrappers);
+        return currentSubscriberWrappers; // Return unchanged state
+      });
+    },
+    [onEmoji]
+  );
```

Comment:
We need an accurate representation of the `subscriberWrappers`. We use the setState function to check the current state to pass it into the `onEmoji` handler.